### PR TITLE
Add pam_limits module to ansible : modify PAM limits for Linux

### PR DESCRIPTION
**Pull request by usawa**
I coded a pam_limits ansible module to modify the /etc/security/limits.conf file, for Linux. It's common to all linux distros. 
If rule is absent, it's added to file
If rule is present, it's modified
use_max and use_min permits to take min or max value between the old or new value.

### CODE REVIEW

**Comment by SiegfriedEhret** [comment]
:+1: that's a good idea. Please merge this !

**Comment by prune998** [comment]
working great ! I needed exactly this !
Thanks

**Comment by mateusduboli** [comment]
Could you add the [ '-' ] option to the type option, it enables setting both hard and soft limits in one statement. The docs are here.
http://linux.die.net/man/5/limits.conf

**Comment by usawa** [comment]
Note : the module doesn't deal with :

"All items support the values -1, unlimited or infinity indicating no limit, except for priority and nice."

I'll work on it when I'll have some time.

File: system/pam_limits.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+import os
+import os.path
+import shutil
+import re
+
+DOCUMENTATION = '''
+---
+module: pam_limits
+version_added: "historical"
```
Comment (bcoca) [inline system/pam_limits.py]: 2.0 should be version

File: system/pam_limits.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    description:
+      - A username, @groupname, wildcard, uid/gid range.
+    required: true
+    default: null
+  limit_type:
+    description:
+      - Limit type : hard or soft.
+    required: true
+    choices: [ "hard", "soft" ]
+    default: null
```
Comment (bcoca) [inline system/pam_limits.py]: required fields do not have defaults

File: system/pam_limits.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    if not found:
+        changed = True
+        new_limit = domain + "\t" + limit_type + "\t" + limit_item + "\t" + str(new_value) + new_comment + "\n"
+        message = new_limit
+        nf.write(new_limit)
+
+    f.close()
+    nf.close()
+
+    # Copy tempfile to newfile
```
Comment (bcoca) [inline system/pam_limits.py:221]: use atomic_move function to put file into place

**Comment by bcoca** [comment]
I'm tempted to just say 'use template module instead' and you get to also use limits.d, but this has some nice features in it.

**Comment by usawa** [comment]
I'll correct problems pointed in your comments. But for the last one, no, we can't use a template module for pam limits, because he template will totally remove the old content. Multiple playbooks/roles must change only specified values, not all content. It should be possible to use something like lineinfile module, but without control on the values.

**Comment by gregdek** [comment]
@usawa happy to merge this if you fix the issues pointed out by @bcoca.

File: system/pam_limits.py (hunk 2 above)
Comment (usawa) [inline system/pam_limits.py]: Can you explain ? Just remove "default" line from documentation ?

File: system/pam_limits.py (hunk 3 above)
Comment (usawa) [inline system/pam_limits.py:221]: done !

**Comment by usawa** [comment]
@gregdek I floowed your comments. It should be ok now, just test it.

A little test playbook :

## ▽
- hosts: localhost
  sudo: yes
  tasks:
  - name: "Pam limits"
    pam_limits: domain="{{ item.domain }}" limit_type="{{ item.limit_type }}" limit_item="{{ item.limit_item }}" value="{{ item.value }}" use_min=yes
    with_items:
    - { domain: "joe", limit_type: "soft", limit_item: "nofile", value: "10000" }
    - { domain: "joe", limit_type: "hard", limit_item: "nofile", value: "20000" }

**Comment by gregdek** [comment]
@bcoca re-review pls?

**Comment by eeroniemi** [comment]
I use this module on production, and it works great.

File: system/pam_limits.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+     - The M(pam_limits) module modify PAM limits, default in /etc/security/limits.conf.
+       For the full documentation, see man limits.conf(5).
+options:
+  domain:
+    description:
+      - A username, @groupname, wildcard, uid/gid range.
+    required: true
+  limit_type:
+    description:
+      - Limit type : hard or soft.
```
Comment (alxgu) [inline system/pam_limits.py:41]: Can't open the documentation with ansible-doc:

ansible-doc pam_limits
Traceback (most recent call last):
 ...
  File "/usr/bin/ansible-doc", line 112, in get_man_text
    desc = " ".join(opt['description'])
TypeError: sequence item 0: expected string, dict found

Please replace the second colon trough dash or whatever :-)

File: system/pam_limits.py [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+    required: true
+    choices: [ "hard", "soft" ]
+  limit_item:
+    description:
+      - The limit to be set : core, data, nofile, cpu, etc.
```
Comment (alxgu) [inline system/pam_limits.py:46]: Here too
Please replace the second colon trough dash or whatever :-)

**Comment by vamfoom** [comment]
I'm attempting to use the pam_limits module as included in Ansible 2.1 but getting this error:

  TASK [system_configs : pam_limits] *********************************************
  fatal: [10.0.107.1]: FAILED! => {"failed": true, "msg": "module (pam_limits) is missing interpreter line"}
  fatal: [10.0.97.1]: FAILED! => {"failed": true, "msg": "module (pam_limits) is missing interpreter line"}
  fatal: [10.0.70.1]: FAILED! => {"failed": true, "msg": "module (pam_limits) is missing interpreter line"}
        to retry, use: --limit @/etc/ansible/main.retry

What am I missing?  This is the task:
- pam_limits: domain=\* limit_item=nofile limit_type=- value=65536

**Comment by alxgu** [comment]
limit_type could be "hard" or "soft" and not -

**Pull request merged**